test/helpers.bash

```
@@ -81,31 +307,106 @@ destroy-kind() {
+      sleep $retry
+    done
+    echo "Error: Could not create monitoring network after multiple attempts"
+    return 1
+  else
+    # Standard mode - no retries
+    $OCI_BIN network create monitoring
+  fi
 }
 
 setup-opensearch() {
   echo "Setting up open-search"
   # Use version 1 to avoid the password requirement
-  $OCI_BIN run --rm -d --name opensearch --network monitoring --env="discovery.type=single-node" --env="plugins.security.disabled=true" --publish=9200:9200 docker.io/opensearchproject/opensearch:1
-  sleep 10
+  # Retry opensearch container start in CI mode
+  if [[ "$CI_MODE" == "true" ]]; then
+    for retry in 1 2 3; do
+      echo "Starting opensearch container (attempt $retry/3)..."
+      if $OCI_BIN run --rm -d --name opensearch --network monitoring --env="discovery.type=single-node" --env="plugins.security.disabled=true" --publish=9200:9200 docker.io/opensearchproject/opensearch:1; then
+        echo "Opensearch container started successfully"
+        sleep 10
+        return 0
+      fi
+      echo "Failed to start opensearch container, retrying in $retry seconds..."
+      $OCI_BIN rm -f opensearch 2>/dev/null || true
+      sleep $retry
+    done
+    echo "Error: Could not start opensearch container after multiple attempts"
+    return 1
+  else
+    # Standard mode - no retries
+    $OCI_BIN run --rm -d --name opensearch --network monitoring --env="discovery.type=single-node" --env="plugins.security.disabled=true" --publish=9200:9200 docker.io/opensearchproject/opensearch:1
+    sleep 10
+  fi
 }
 
 setup-grafana() {
   export GRAFANA_URL="http://localhost:3000"
   export GRAFANA_ROLE="admin"
   echo "Setting up Grafana"
-  $OCI_BIN run --rm -d --name grafana --network monitoring -p 3000:3000 \
-    --env GF_SECURITY_ADMIN_PASSWORD=${GRAFANA_ROLE} \
-    docker.io/grafana/grafana:latest
-  sleep 10
-  echo "Grafana is running at $GRAFANA_URL"
+  
+  # Retry grafana container start in CI mode
+  if [[ "$CI_MODE" == "true" ]]; then
```

Same

---

Removed CI_MODE variable and retry logic

Replaced complex retry logic with a simpler fail-fast approach
Removed the CI_MODE variable from both [helpers.bash](https://bug-free-meme-pjp7x9769rrg36wjp.github.dev/) and workflow files
Updated comment to explain that tests now fail fast if there are setup issues
Standardized setup functions

Made all setup-* functions follow the same pattern:
Try once to create/start the resource
Print "FATAL:" error message and exit 1 if it fails
Print success message if it succeeds
Applied to setup-prometheus, setup-shared-network, setup-opensearch, and setup-grafana.

Fixed error handling

All error messages now consistently use "FATAL:" prefix
All setup functions now use exit 1 for failures (not return 1)
No more automatic test skipping - only user-controlled via SKIP_*_INSTALL variables
Removed outdated or redundant code

Removed all unnecessary retry logic and fallbacks
Simplified check_metric_value to use continue instead of return 0
Workflow file updates

Removed CI_MODE from the [ci-parallel.yml](https://bug-free-meme-pjp7x9769rrg36wjp.github.dev/) workflow file...